tests/specs/lemmas.k

```diff
@@ -65,6 +72,24 @@ module LEMMAS
     rule WS         ++ .ByteArray => WS  [simplification]
     rule .ByteArray ++ WS         => WS  [simplification]
 
+    // #range
+
+    rule #range(_                       , _, K) => .ByteArray                                                                                                                                                          requires notBool K >Int 0                                                                               [simplification]
+    rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, minInt(K, N -Int L)) ++ #range(M [ N := BUF ], N, K -Int minInt(K, N -Int L))                                                                          requires K  >Int 0 andBool L <Int N                                                                     [simplification]
+   rule #range(M [ N := BUF:ByteArray ], L, K) => 
+          BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ] ++ 
+          #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))
+     requires K  >Int 0 andBool L <=Int N +Int #sizeByteArray(BUF) andBool L >=Int N [simplification]
+
+    rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, K) 
+      requires K >Int 0 
+       andBool L >=Int N +Int #sizeByteArray(BUF) [simplification]
+
+
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
```

Comment:
Formatting here is strange.

Reply:
The first two have weird spacing from before which I can fix. The lemma on line 79 seems hard to format in a pleasing way. Does this make more sense? @ehildenb 
```
    rule #range(_, _, K) => 
         .ByteArray 
      requires notBool K >Int 0 [simplification]

    rule #range(M [ N := BUF:ByteArray ], L, K) => 
         #range(M, L, minInt(K, N -Int L)) ++ #range(M [ N := BUF ], N, K -Int minInt(K, N -Int L)) 
      requires K  >Int 0 andBool L <Int N [simplification]

   rule #range(M [ N := BUF:ByteArray ], L, K) => 
          BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ]
          ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))
     requires K  >Int 0 andBool L <=Int N +Int #sizeByteArray(BUF) andBool L >=Int N [simplification]

    rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, K) 
      requires K >Int 0 
       andBool L >=Int N +Int #sizeByteArray(BUF) [simplification]

    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ], L, K ) =>
         #range( M                          [ N2 := BUF2           ], L, K )
      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]

```

Reply:
I mostly was referring to how 3 spaces was used for some of this instead of 4. I do usually prefer associative operators on the line before the next one (like you've done with `++` in the third one).

Reply:
The spacing should be fixed.